sdk/src/main/java/com/hedera/hashgraph/sdk/Network.java

```diff
@@ -36,13 +41,114 @@
+        network.put("35.204.86.32:50211", new AccountId(12));
+        network.put("35.234.132.107:50211", new AccountId(13));
+        network.put("35.236.2.27:50211", new AccountId(14));
+        network.put("35.228.11.53:50211", new AccountId(15));
+        network.put("34.91.181.183:50211", new AccountId(16));
+        network.put("34.86.212.247:50211", new AccountId(17));
+        network.put("172.105.247.67:50211", new AccountId(18));
+        network.put("34.89.87.138:50211", new AccountId(19));
+        network.put("34.82.78.255:50211", new AccountId(20));
+
+        return new Network(executor, network).setNetworkName(NetworkName.MAINNET);
+    }
+
+    static Network forTestnet(ExecutorService executor) {
+        var network = new HashMap<String, AccountId>();
+        network.put("0.testnet.hedera.com:50211", new AccountId(3));
+        network.put("1.testnet.hedera.com:50211", new AccountId(4));
+        network.put("2.testnet.hedera.com:50211", new AccountId(5));
+        network.put("3.testnet.hedera.com:50211", new AccountId(6));
+        network.put("4.testnet.hedera.com:50211", new AccountId(7));
+
+        return new Network(executor, network).setNetworkName(NetworkName.TESTNET);
+    }
+
+    static Network forPreviewnet(ExecutorService executor) {
+        var network = new HashMap<String, AccountId>();
+        network.put("0.previewnet.hedera.com:50211", new AccountId(3));
+        network.put("1.previewnet.hedera.com:50211", new AccountId(4));
+        network.put("2.previewnet.hedera.com:50211", new AccountId(5));
+        network.put("3.previewnet.hedera.com:50211", new AccountId(6));
+        network.put("4.previewnet.hedera.com:50211", new AccountId(7));
+
+        return new Network(executor, network).setNetworkName(NetworkName.PREVIEWNET);
+    }
+
+    @Nullable
+    NetworkName getNetworkName() {
+        return networkName;
+    }
+
+    Network setNetworkName(NetworkName networkName) {
+        this.networkName = networkName;
+
+        for (var node : nodes) {
+            node.setNetworkName(networkName);
+        }
+
+        if (!nodeAddressBooks.containsKey(networkName)) {
+            var resource = "";
+
+            switch (networkName) {
+                case MAINNET:
+                    resource = "./addressbook/mainnet.pb";
+                    break;
+                case TESTNET:
+                    resource = "./addressbook/testnet.pb";
+                    break;
+                case PREVIEWNET:
+                    resource = "./addressbook/previewnet.pb";
+                    break;
+            }
+
+            nodeAddressBooks.put(networkName, readAddressBookResource(resource));
+        }
+
+        return this;
+    }
+
+    static Map<AccountId, NodeAddress> readAddressBookResource(String fileName) {
+        try {
+            var contents = ByteStreams.toByteArray(Objects.requireNonNull(Client.class.getClassLoader().getResourceAsStream(fileName)));
```

Comment:
This inputstream is not properly closed in a try with resources. It can also fail due to multiple classloaders. Suggest using Guava:

```java
        try (InputStream inputStream = Resources.getResource(fileName).openStream()) {
            var contents = ByteStreams.toByteArray(inputStream);
```

Reply:
Thanks for the suggestion, I wasn't aware of this..